packages/@tinacms/graphql/src/resolver/index.ts

```diff
@@ -124,9 +112,12 @@ export class Resolver {
       )
 
       return {
-        __typename: NAMER.documentTypeName([rawData._collection]),
+        __typename: collection.fields
+          ? NAMER.documentTypeName(collection.namespace)
+          : NAMER.documentTypeName(template.namespace),
```

Comment:
The `template` namespace will be something like `['post', 'blogPost']`, whereas collection is just `['post']`. My comments from the `build/index.ts` are relevant here, because we've sort of flattened the result here for collections with `templates`.